src/LuaTransformer.ts

```diff
@@ -3664,16 +3664,32 @@ export class LuaTransformer {
             flags |= tstl.FunctionExpressionFlags.Inline;
         }
 
-        const [transformedBody] = this.transformFunctionBody(node.parameters, body, spreadIdentifier);
+        const [transformedBody, scope] = this.transformFunctionBody(node.parameters, body, spreadIdentifier);
 
-        return tstl.createFunctionExpression(
+        const functionExpression = tstl.createFunctionExpression(
             tstl.createBlock(transformedBody),
             paramNames,
             dotsLiteral,
             spreadIdentifier,
             flags,
             node
         );
+
+        if (ts.isFunctionExpression(node) && node.name && scope.referencedSymbols) {
```

Comment:
Would probably not be a bad idea if y ou added a comment above this block saying this handles named function assignments.